server/src/app.ts

```diff
@@ -49,21 +55,25 @@ export const createApp = (): express.Application => {
   app.use(cors());
   app.use(middleware.handle(i18next));
   app.use(express.static(filePath));
+  app.use((req: ReqI18n, res, next) => {
+    const { JWT_SECRET } = process.env;
+    req.appSecret = JWT_SECRET;
+    next();
+  });
 
   app.set('views', path.join(__dirname, '../html'));
   app.engine('html', ejs.renderFile);
   app.set('view engine', 'html');
 
-  app.get('/reset_password/:email/:hashed/:password', async (req: ReqI18n, res) => {
-    const email = qs.unescape(req.params.email);
-    const hashed = qs.unescape(req.params.hashed);
+  app.get('/reset_password/:token/:password', async (req: ReqI18n, res) => {
+    const token = qs.unescape(req.params.token);
     const randomPassword = qs.unescape(req.params.password);
 
     try {
-      const validated = await validateCredential(email, hashed);
-      if (validated) {
+      const validated = verify(token, req.appSecret) as VerificationToken;
+      if (validated && validated.email && validated.type === 'findPassword') {
```

Comment:
How about using [optional-chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) in this case?

Reply:
Resolved in 7691ed5